src/lib/ptree/ptree.lua

```diff
@@ -295,6 +297,38 @@ function Manager:compute_scheduling_for_worker(id, app_graph)
    return ret
 end
 
+local counters = {active={}, archived={}}
+
+local function is_counter (name)
+   return lib.basename(name):match("%.counter$")
+end
+local function read_counter (name)
+   if not counters.active[name] then counters.active[name] = counter.open(name) end
+   return counter.read(counters.active[name])
+end
+local function update_counter (name, val)
+   local c = assert(counters.active[name])
+   counter.add(c, val)
+end
+local function create_counter (name, val)
+   counters.active[name] = counter.create(name, val)
+end
+local function archive_counter (name)
+   local c = assert(counters.active[name])
+   counter.delete(c)
+   counters.archived[name] = c
```

Comment:
To archive a counter, you should sample its current value and add it to the corresponding archived counter.  After `counter.delete`, the memory is no longer mapped, so it is no longer accessible.